src/objects/movers/PathPlayerMover.cpp

```diff
@@ -2,24 +2,32 @@
 
 #include "PathPlayerMover.h"
 
-void PathPlayerMover::move(float deltaTime) {
-    // calculate vec3 to move towards nextPoint
-    glm::vec3 flatMovingTowards = movingTowards;
-    flatMovingTowards.y = 1.0f;
-    glm::vec3 flatPlayerPosition = player.GetPosition();
-    flatPlayerPosition.y = 1.0f;
-    glm::vec3 movePoint = flatMovingTowards - flatPlayerPosition;
-
-    player.MoveIn(movePoint, deltaTime);
-
-    if (abs(player.GetPosition().x - movingTowards.x ) < 0.5f &&
-        abs(player.GetPosition().z - movingTowards.z) < 0.5f &&
-        movingTowards.x == path.getPath().at(pointer).x
-        ) {
-        if (pointer + 1 < path.getPath().size()) {
-            pointer++;
-            movingTowards = path.getPath().at(pointer);
+static constexpr const float AT_TARGET = std::numeric_limits<float>::epsilon();
+
+void PathPlayerMover::move(float distance) {
+    while (distance > 0) {
+
+        const glm::vec2 flatCurrent{ player.GetPosition().x, player.GetPosition().z };
+
+        glm::vec2 flatTarget{ movingTowards.x, movingTowards.z };
+        float distToTarget = glm::distance(flatCurrent, flatTarget);
+        if (distToTarget < AT_TARGET) {
+            if(pointer >= path.getPath().size() - 1)
+                return; // we are at the end of the path
+            movingTowards = path.getPath().at(++pointer);
+ 
+            flatTarget = glm::vec2{ movingTowards.x, movingTowards.z };
+            distToTarget = glm::distance(flatCurrent, flatTarget);
```

Comment:
If (for whatever reason) I have 2 identical values in my `path` property, let's say:
- 0: glm::vec3(1.5f,0.0f,-2.5f)
- 1: glm::vec3(1.5f,0.0f,-2.5f)
and we just update `movingTowards` to point from 0 to 1, then the `distToTarget` _may_ become 0. In that case this

```c++
        const glm::vec2 dir = (flatTarget - flatCurrent) / distToTarget;
```

would be null, right?

Should we make sure `distToTarget > std::numeric_limits<float>::epsilon()` ?

Reply:
Agree. It is better to make sure we don't have duplicate waypoints 💪